### PR TITLE
Fix transactions commands

### LINKED ISSUE #99: deletet deletes transactions if im in the person list page
If i first `listt INDEX`, the `list`, the transaction list is still saved. Transaction list should be cleared upon using list command.

This should also fix the other transaction related commands producing undesired behaviours in the person list page.

### LINKED ISSUE #116: `findt` only in person page
Currently, we says in the command format and user guide that you should only use `findt` in person page, but don't prevent users from doing it.
Can use `!model.getIsTransactionView` to verify the current view is person page before the command is executed.
Also applicable for `addt`.

### LINKED ISSUE #120: Change `deletet` message
Screenshot 2024-10-28 at 11 17 40 PM

Currently, there is no indication that the user should navigate to a transaction list
before using deletet. The error message could be updated to show that.

### PR DESCRIPTION
**Pull request by beginner-web-developer**
This PR fixes:

- Clears transactions list when user requests to `list`
- Updates code to throw exception when the user access the commands in the wrong view (transaction vs person view).

Fixes #99 , fixes #116 , fixes #120

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FindTransactionCommand.java [hunk 1]
```diff
@@ -43,14 +43,19 @@ public FindTransactionCommand(Index presonIndex, TransactionContainsKeywordsPred
...
+
         List<Person> lastShownList = model.getFilteredPersonList();
         if (personIndex.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
         Person targetPerson = lastShownList.get(personIndex.getZeroBased());
         List<Transaction> targetTransactions = targetPerson.getTransactions();
         model.updateFilteredPersonList(new IsSelectedPredicate(model, personIndex));
-        model.setViewTransactions(true);
+        //model.setViewTransactions(true);
```
Comment (reginateh) [inline src/main/java/seedu/address/logic/commands/FindTransactionCommand.java:58]: Why remove this? Shouldn't `findt` switch the view from person to transaction?

File: src/main/java/seedu/address/logic/commands/ListTransactionCommand.java [hunk 2]
```diff
@@ -33,14 +33,19 @@ public ListTransactionCommand(Index index) {
...
+
         List<Person> lastShownList = model.getFilteredPersonList();
         if (index.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
 
         Person selected = lastShownList.get(index.getZeroBased());
         model.updateFilteredPersonList(new IsSelectedPredicate(model, index));
-        model.setViewTransactions(true);
+        //model.setViewTransactions(true);
```
Comment (reginateh) [inline src/main/java/seedu/address/logic/commands/ListTransactionCommand.java:48]: Same as above

Comment (beginner-web-developer) [reply src/main/java/seedu/address/logic/commands/ListTransactionCommand.java:48]: i have moved that logic to model#updateTransactionList()

File: src/main/java/seedu/address/logic/commands/FindTransactionCommand.java (hunk 1 above)
Comment (reginateh) [reply src/main/java/seedu/address/logic/commands/FindTransactionCommand.java:58]: Oh I see. `updateTransactionList` set the view to transaction.

**Comment by reginateh** [comment]
LGTM

**Comment by beginner-web-developer** [comment]
wait i should remove the comments

**Pull request merged**